realm-annotations-processor/src/main/java/io/realm/processor/Utils.java

```diff
@@ -73,6 +76,38 @@ public static boolean isString(VariableElement field) {
     }
 
     /**
+     * Return true if a given type string represents a primitive type.
+     */
+    public static boolean isPrimitiveType(String typeString) {
+        if (typeString == null) {
+            return false;
+        }
+        return typeString.equals("byte") || typeString.equals("short") || typeString.equals("int") ||
+                typeString.equals("long") || typeString.equals("float") || typeString.equals("double") ||
+                typeString.equals("boolean") || typeString.equals("char");
+    }
+
+    /**
+     * Return true if a field is a type of primitive types.
+     */
+    public static boolean isPrimitiveType(VariableElement field) {
+        if (field == null) {
+            return false;
+        }
+        return field.asType().getKind().isPrimitive();
+    }
+
+    /**
+     * Returns true if a field is of type "byte[]", false otherwise.
```

Comment:
We don't support `Byte[]`. Processor will throw a `Error:(25, 8) error: Type java.lang.Byte[] of field coloooo is not supported`